content/blog/blog-2020-07-01-kudo-dependencies.md

```diff
@@ -0,0 +1,93 @@
+
+## Dependency Parametrization
+
+Overall we wanted to encourage operator composition by providing a way of operator encapsulation. In other words, operator users should not be allowed to arbitrarily modify the parameters of embedded operator instances. The higher-level operator should define all parameters that its direct dependency operators need. If a child operator needs to be parametrised a parameter file can be specified using the `parameterFile` field. Let's take a look at an example. Suppose, we have two operators (_parent_ and _child_) and _child_ has a required parameter `PASSWORD` which is empty by default. First, the parent operator needs to specify a  parameter file for the child and reference it in the corresponding `KudoOperator` task:
+
+```yaml
+tasks:
+  - name: deploy-child
+    kind: KudoOperator
+    spec:
+      package: child-operator
+      parameterFile: child-params.yaml 
+```
+
+`child-params.yaml` is located in the parents _template_ folder along with other template files:
+
+```yaml
+# parent/templates/child-params.yaml
+PASSWORD: {{ .Params.CHILD_PASSWORD }}
+```
+
+Child `PASSWORD` value references the parent `CHILD_PASSWORD` parameter that can look like:
+
+```yaml
+# parent/params.yaml
+apiVersion: kudo.dev/v1beta1
+parameters:
+  - name: CHILD_PASSWORD
+    displayName: "child password"
+    description: "password for the underlying instance of child operator"
+    required: true
+```
+
+When installing the _parent_ operator the user then has to define the `CHILD_PASSWORD` as usual:
+
+```shell script
+$ kubectl kudo install parent -p CHILD_PASSWORD=secret
+```
+
+Note, that the _parent_ operator may decide to provide a sensible default or even to hardcode the password and not expose it at all to the end user.
+
+## Summary
+
+Obviously, there is more to dependencies than the right installation order. One area that KEP-29 deliberately avoided is life-cycle dependencies. It is easy to imagine a situation when a new operator (e.g Kafka) may want to depend on the existing Zookeeper instance. However, such life-cycle dependencies presents major challenges e.g. what happens when Zookeeper is removed? What happens when Zookeeper is upgraded, and the new version is incompatible with the current Kafka Instance? How can we ensure the compatibility? Anyway, we want to gather feedback before we address these issues so ping us if you have any ideas. For more information and implementation details, take a look at the [KEP-29](https://github.com/kudobuilder/kudo/blob/main/keps/0029-operator-dependencies.md), updated [Flink demo](https://github.com/kudobuilder/operators/pull/279) or [dependencies documentation](https://kudo.dev/docs/developing-operators/tasks.html#kudooperator-task).
```

Comment:
I would remove "Anyway"... it seems dismissive... the sentence works without it.